assets/js/components/Button.js

```diff
@@ -21,72 +21,69 @@
  */
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import useMergedRef from '@react-hook/merged-ref';
+
 /**
  * WordPress dependencies
  */
-import { Component, createRef } from '@wordpress/element';
+import { forwardRef, useRef, useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import { MDCRipple } from '../material-components';
 
-class Button extends Component {
-	constructor( props ) {
-		super( props );
-		this.buttonRef = createRef();
-	}
-
-	componentDidMount() {
-		new MDCRipple( this.buttonRef.current );
-	}
+const Button = forwardRef( ( {
+	children,
+	href,
+	text,
+	className,
+	danger,
+	disabled,
+	target,
+	icon,
+	trailingIcon,
+	ariaHaspopup,
+	ariaExpanded,
+	ariaControls,
+	...extraProps
+}, ref ) => {
+	const buttonRef = useRef( null );
+	useEffect( () => {
+		MDCRipple.attachTo( buttonRef.current );
+	}, [ buttonRef.current ] );
 
-	render() {
-		const {
-			children,
-			href,
-			text,
-			className,
-			danger,
-			disabled,
-			target,
-			icon,
-			trailingIcon,
-			ariaHaspopup,
-			ariaExpanded,
-			ariaControls,
-			...extraProps
-		} = this.props;
+	// Use a button if disabled, even if a href is provided to ensure expected behavior.
+	const SemanticButton = ( href && ! disabled ) ? 'a' : 'button';
 
-		// Use a button if disabled, even if a href is provided to ensure expected behavior.
-		const SemanticButton = ( href && ! disabled ) ? 'a' : 'button';
+	return (
+		<SemanticButton
+			className={ classnames(
+				'mdc-button',
+				className,
+				{
+					'mdc-button--raised': ! text,
+					'mdc-button--danger': danger,
+				}
+			) }
+			href={ disabled ? undefined : href }
+			ref={ useMergedRef( ref, buttonRef ) }
```

Comment:
This looks off to me. Even if this technically works, we should put all our hook calls at the top to avoid potential errors in the future and make it more obvious what hooks are being used.

Reply:
Updated @aaemnnosttv